respond/json.go

```diff
@@ -0,0 +1,23 @@
+package respond
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Jille/convreq/internal"
+)
+
+type respondJSON struct {
+	data interface{}
+}
+
+// Respond implements convreq.HttpResponse.
+func (rj respondJSON) Respond(w http.ResponseWriter, r *http.Request) error {
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
Let's only set this header if it isn't set yet (for example if people want to use text/json)